RecoHGCal/TICL/plugins/TrackstersProducer.cc

```diff
@@ -179,8 +179,19 @@ void TrackstersProducer::produce(edm::Event& evt, const edm::EventSetup& es) {
   std::copy(
       std::begin(original_layerclusters_mask), std::end(original_layerclusters_mask), std::back_inserter(*output_mask));
 
-  // Mask the used elements, accordingly
-  for (auto const& trackster : *result) {
+  for (auto& trackster : *result) {
+    if (itername_ == "TrkEM")
+      trackster.setIteration(ticl::Trackster::TRKEM);
```

Comment:
it looks like `itername_` is known at the configuration step.
please move this string->enum remapping to the constructor where `iterIndex_` can be computed more effectively.

Reply:
@slava77 `itername_` is known at the configuration step but the `result` vector is not.
Shall I define an auxiliary variable to store the string->enum remapping in the constructor, and assign it to the `trackster` as in the current code?

Reply:
sure; that's what I meant : to compute `IterationIndex iterIndex_` in the constructor and use it here in this loop directly.

Reply:
Done, thanks.